databuilder/query_model.py

```diff
@@ -49,7 +50,7 @@
 # type without specifying what that type has to be
 T = TypeVar("T")
 Numeric = TypeVar("Numeric", int, float)
-Comparable = TypeVar("Comparable", int, float, str, date)
+Comparable = TypeVar("Comparable", int, float, str, date, BaseCode)
```

Comment:
Commenting here to respond to the commit message:

> This isn't something that researchers will ever want to do, but it
> allows us to use code columns for tie-breaking sorts to ensure sort
> stability.

I think users may want to do this if they want to ensure exactly the same output between runs.  I think we ensure sort stability in parts of cohort-extractor for this reason.

Reply:
My claim is that researchers don't every really want to sort by code for its own sake. They may want to have sort stability, but this isn't the commit that gives that.

Reply:
Makes sense.